lib/src/SlidingWindowRateLimiter.cc

```diff
@@ -0,0 +1,58 @@
+#include "SlidingWindowRateLimiter.h"
+#include <assert.h>
+
+using namespace drogon;
+
+SlidingWindowRateLimiter::SlidingWindowRateLimiter(
+    size_t capacity,
+    std::chrono::duration<double> timeUnit)
+    : capacity_(capacity),
+      unitStartTime_(std::chrono::steady_clock::now()),
+      lastTime_(unitStartTime_),
+      timeUnit_(timeUnit)
+{
+}
+// implementation of the sliding window algorithm
+bool SlidingWindowRateLimiter::isAllowed()
+{
+    auto now = std::chrono::steady_clock::now();
+    unitStartTime_ =
+        unitStartTime_ +
+        std::chrono::duration_cast<decltype(unitStartTime_)::duration>(
+            std::chrono::duration<double>(
+                static_cast<double>((uint64_t)(
```

Comment:
Why cast to `uint64` then back to `double`?

Reply:
Make the unitStartTime_ is earlier than the current time.